File: python_code/common_codes/a10_compare_all.py
import numpy as np
import matplotlib.pyplot as plt


def a10_compare_all(All, name):
    a = np.arange(1, All.shape[0] + 1)
    s = plt.figure()
    s.set_facecolor('white')
    colors = ['r', 'g', 'b', 'y', 'c', 'm', 'k']

    for i in range(All.shape[1]):
        plt.plot(a, All[:, i], color=colors[i % len(colors)], marker='o',
                 linestyle=':', markerfacecolor=colors[i % len(colors)],
                 label=name[i] if i < len(name) else f'Method {i+1}')

    plt.legend()
    plt.axhline(y=50, color='g', linestyle='--', alpha=0.5)
    plt.axhline(y=100, color='g', linestyle='--', alpha=0.5)
    plt.axhline(y=150, color='g', linestyle='--', alpha=0.5)
    plt.xlabel('Sample ID')
    plt.ylabel('WQI value')
    plt.grid(True, alpha=0.3)
    plt.tight_layout()
    return s

File: common_codes/a10_compare_all.m
function a10_compare_all(All,name)
a=1:size(All,1);
s=figure()
set(gcf,'color','w')
h1=plot(a',All);
baseColors = {'r', 'g', 'b', 'y', 'c', 'm', 'k'};
use=baseColors(1:numel(h1));
[h1.Color]=deal(use{:});%"r","g","b","c","m"
[h1.Marker]=deal('o');
[h1.LineStyle]=deal(':');
[h1.MarkerFaceColor] = deal(use{:});
l=legend(h1);
l.String=cellstr(strcat(name));
l.AutoUpdate="off";
yline([50 100 150],'--g')
xlabel('Sample ID');ylabel('WQI value');
copygraphics(s,'Resolution',600,'BackgroundColor','white','Padding','tight','PreserveAspectRatio','on')
disp('****')